core/src/test/java/dk/alexandra/fresco/framework/network/TestKryoNetNetwork.java

```diff
@@ -189,22 +191,23 @@ public void run() {
   public void testKryoNetConnectInterrupt() {
     Map<Integer, Party> parties = new HashMap<>();
     List<Integer> ports = getFreePorts(2);
-    parties.put(1, new Party(1, "localhost", ports.get(0)));
-    parties.put(2, new Party(2, "localhost", ports.get(1)));    
+    parties.put(1, new Party(1, "localhost", 9000));
+    parties.put(2, new Party(2, "localhost", ports.get(1)));
+    List<Exception> exs = new ArrayList<>();
     Thread t1 = new Thread(new Runnable() {
       
       @Override
       public void run() {
         NetworkConfiguration conf = new NetworkConfigurationImpl(1, parties);
         KryoNetNetwork network = null;
-        try{
+        try {
           network = new KryoNetNetwork(conf);
           Assert.fail("Should not be able to connect");
```

Comment:
Small suggestion - this could be a RuntimeException (for clarity), the Assert.fail throws a more uncontrolled exception

Reply:
Redid all asserts within threads since this did not actually cause the test to fail if they failed.